src/wallet/wallet.cpp

```diff
@@ -1237,7 +1237,7 @@ CAmount CWalletTx::GetCredit(const isminefilter& filter) const
 CAmount CWalletTx::GetImmatureCredit(bool fUseCache, const isminefilter& filter) const
 {
     LOCK(cs_main);
-    if ((IsCoinBase() || IsCoinStake()) && GetBlocksToMaturity() > 0 && IsInMainChain()) {
+    if (GetBlocksToMaturity() > 0 && IsInMainChain()) {
```

Comment:
This can unified in just a single method call.

1) `GetBlocksToMaturity` is calling internally `GetDepthInMainChain`.
2) `IsInMainChain` is calling internally `GetDepthInMainChain`.